packages/core/src/render3/util/discovery_utils.ts

```diff
@@ -181,18 +184,30 @@ export function getInjectionTokens(element: Element): any[] {
  * </my-app>
  * ```
  * Calling `getDirectives` on `<button>` will return an array with an instance of the `MyButton`
- * directive that is associated with the DOM element.
+ * directive that is associated with the DOM node.
  *
  * Calling `getDirectives` on `<my-comp>` will return an empty array.
  *
- * @param element DOM element for which to get the directives.
- * @returns Array of directives associated with the element.
+ * @param node DOM node for which to get the directives.
+ * @returns Array of directives associated with the node.
  *
  * @publicApi
  * @globalApi ng
  */
-export function getDirectives(element: Element): {}[] {
-  const context = loadLContext(element)!;
+export function getDirectives(node: Node): {}[] {
+  if (node instanceof Text) {
+    return [];
+  }
+  const context = loadLContext(node, false);
+
+  if (context === null) {
+    return [];
+  }
+
+  const tView = context.lView[TVIEW];
+  if (!tView || !tView.data[context.nodeIndex]) {
+    return [];
+  }
```

Comment:
```suggestion
  if (!tView?.data[context.nodeIndex]) {
    return [];
  }
```
Though this should never happen and LView should always have associated TView (i.e. we can probably remove `?` here).